### PR TITLE
Connect compiler to endpoints

### PR DESCRIPTION
**Pull request by JRojas999**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package org.fundacion.jala.converter.view.compiler;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.geom.Line2D;
+import java.awt.geom.Point2D;
+
+class CloseButton extends JButton {
+
+    public void paintComponent(Graphics g) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java:28]: javadoc missing

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        super.paintComponent(g);
+
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java]: unnecesary newLine

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 3]
```diff
@@ -83,4 +83,11 @@ public void removeUpdate(final DocumentEvent de) {
         textCodeArea.setRowHeaderView(lineCode);
         add(textCodeArea, BorderLayout.CENTER);
     }
+    public String getText() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:86]: missing javadoc

Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:93]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java [hunk 4]
```diff
@@ -83,4 +83,11 @@ public void removeUpdate(final DocumentEvent de) {
...
+        return codeArea.getText();
+    }
+
+    public JTextArea getCodeArea() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java [hunk 5]
```diff
@@ -65,18 +66,11 @@ protected CompilerButtons() {
...
-                pathFile = "";
-            } else {
-                pathFile = fileName.getAbsolutePath();
-            }
-        }
+        
+        System.out.println("Hola bebe");
+    }
+
+    public JButton getRunButton() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java:74]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 6]
```diff
@@ -9,9 +9,22 @@
...
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
+import javax.swing.*;
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: You should import each library and do not use *.

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 7]
```diff
@@ -9,9 +9,22 @@
...
+import java.awt.*;
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: You should import each library and do not use *.

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 8]
```diff
@@ -52,5 +82,35 @@ public CompilerInterface() {
         panelConstraint.gridheight = 2;
         panelConstraint.gridwidth = 4;
         add(consoleOutput, panelConstraint);
+
+        buttonsCompiler.getRunButton().addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:89]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java [hunk 9]
```diff
@@ -34,4 +34,8 @@ protected Console() {
         setLayout(new BorderLayout());
         add(console, BorderLayout.CENTER);
     }
+
+    public JTextArea getConsole() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:44]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            String temp = getTitleAt(index);
+            if (temp.equals(title)) {
+                removeTabAt(index);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public CodeTextArea getSelectedPane() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:85]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+        return (CodeTextArea) getSelectedComponent();
+    }
+
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: unnecesary newLine

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+        JLabel label=new JLabel(title);
+        label.setFont(new Font("Barlow", 0, 11));
+        pnl.add(label);
+        pnl.add(tabButton);
+
+        setTabComponentAt(getTabCount() - 2, pnl);
+        setSelectedIndex(getTabCount() - 2);
+    }
+
+    public boolean removeTap(String title) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:69]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+        button.addActionListener(this);
+
+        add(new JPanel());
+        setTabComponentAt(getTabCount() - 1, button);
+        contador = 0;
+    }
+    CodeTextArea codeArea = new CodeTextArea();
+
+    @Override
+    public void actionPerformed(ActionEvent e) {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:44]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 14]
```diff
@@ -0,0 +1,66 @@
+package org.fundacion.jala.converter.view.compiler;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class ProjectTab extends JTabbedPane implements ActionListener {
+    public static int contador;
+    public void start() {
```
Comment (Raymundogs) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:30]: missing javadoc

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java [hunk 15]
```diff
@@ -65,18 +66,11 @@ protected CompilerButtons() {
...
-        if (result != JFileChooser.CANCEL_OPTION) {
-            File fileName = fileChooser.getSelectedFile();
-            if ((fileName == null) || (fileName.getName().equals(""))) {
-                pathFile = "";
-            } else {
-                pathFile = fileName.getAbsolutePath();
-            }
-        }
+        
+        System.out.println("Hola bebe");
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java]: This line is not necessary

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 16]
```diff
@@ -9,9 +9,22 @@
...
+import org.apache.http.HttpEntity;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This blank line it's not necessary

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 17]
```diff
@@ -27,6 +40,21 @@ public CompilerInterface() {
         consoleOutput = new Console();
         langButtons = new LanguageButtons();
         codeArea = new CodeTextArea();
+
+        ProjectTab projectTab = new ProjectTab();
+        projectTab.setFont(new Font("Barlow", 0, 11));
+
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This blank line it's not necessary

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 18]
```diff
@@ -27,6 +40,21 @@ public CompilerInterface() {
...
+        projectTab.add(codeArea);
+        JPanel pnl = new JPanel();
+        pnl.setLayout(new FlowLayout());
+        pnl.setOpaque(false);
+        JLabel label=new JLabel("Main");
+        label.setFont(new Font("Barlow", 0, 11));
+        pnl.add(label);
+        projectTab.setTabComponentAt(projectTab.getTabCount() - 1, pnl);
+
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This blank line it's not necessary

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 19]
```diff
@@ -52,5 +82,35 @@ public CompilerInterface() {
...
         add(consoleOutput, panelConstraint);
+
+        buttonsCompiler.getRunButton().addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                String url = "http://localhost:8080/api/compilePython";
+                HttpPost httpPost = new HttpPost(url);
+
+                String code1 = projectTab.getSelectedPane().getText();
+                System.out.println(code1);
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This line it's not necessary

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 20]
```diff
@@ -52,5 +82,35 @@ public CompilerInterface() {
...
+                try {
+                    httpPost.setEntity(multipart);
+                    httpPost.setHeader("Authorization", "Bearer " + findUserById(1).getToken());
+                    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+                    CloseableHttpResponse response = httpClient.execute(httpPost);
+                    HttpEntity responseEntity = response.getEntity();
+                    String sResponse = EntityUtils.toString(responseEntity, "UTF-8");
+                    consoleOutput.getConsole().setText(sResponse);
+                    System.out.println(sResponse);
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This line it's not necessary

File: src/main/java/org/fundacion/jala/converter/view/MainInterface.java [hunk 21]
```diff
@@ -70,4 +70,8 @@ public void initInterface() {
         setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
         setVisible(true);
     }
+
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/MainInterface.java:76]: Line 24: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 22]
```diff
@@ -40,7 +68,9 @@ public CompilerInterface() {
         panelConstraint.gridy = 1;
         panelConstraint.gridheight = 2;
         panelConstraint.gridwidth = 4;
-        add(codeArea, panelConstraint);
+
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java]: This blank line should be removed

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java [hunk 23]
```diff
@@ -52,5 +82,35 @@ public CompilerInterface() {
...
+                String code1 = projectTab.getSelectedPane().getText();
+                System.out.println(code1);
+
+                MultipartEntityBuilder builder;
+                builder = MultipartEntityBuilder.create();
+                builder.addTextBody("code", code1, ContentType.TEXT_PLAIN);
+                HttpEntity multipart;
+                multipart = builder.build();
+                try {
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:106]: Logs are missing in this try - catch

File: src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java [hunk 24]
```diff
@@ -0,0 +1,23 @@
+package org.fundacion.jala.converter.view.compiler;
+
+import javax.swing.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java [hunk 25]
```diff
@@ -0,0 +1,23 @@
...
+import java.awt.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/PlusButton.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 26]
```diff
@@ -0,0 +1,66 @@
+package org.fundacion.jala.converter.view.compiler;
+
+import javax.swing.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 27]
```diff
@@ -0,0 +1,66 @@
...
+import java.awt.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java [hunk 28]
```diff
@@ -0,0 +1,66 @@
...
+import java.awt.event.ActionListener;
+
+public class ProjectTab extends JTabbedPane implements ActionListener {
+    public static int contador;
+    public void start() {
+        PlusButton button;
+        button=new PlusButton();
+        button.setPreferredSize(new Dimension(20,20));
+        button.addActionListener(this);
+
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java]: There should be no blank lines between the lines of code in this class

File: src/main/java/org/fundacion/jala/converter/view/MainInterface.java (hunk 21 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/MainInterface.java:76]: It is already done

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java (hunk 1 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java:28]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java (hunk 3 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CodeTextArea.java:93]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java (hunk 5 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CompilerButtons.java:74]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java (hunk 23 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:106]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 13 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:44]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 12 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:69]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 10 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:85]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java (hunk 8 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/CompilerInterface.java:89]: It is not necesary

File: src/main/java/org/fundacion/jala/converter/view/compiler/Console.java (hunk 9 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/Console.java:44]: There it is

File: src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java (hunk 14 above)
Comment (JRojas999) [reply src/main/java/org/fundacion/jala/converter/view/compiler/ProjectTab.java:30]: There it is

**Review by edsonarios — APPROVED** [review]

File: src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java [hunk 29]
```diff
@@ -0,0 +1,41 @@
...
+ * Copyright (c) 2021 Fundacion Jala.
+ *
+ * This software is the confidential and proprietary information of Fundacion Jala
+ * ("Confidential Information"). You shall not disclose such Confidential
+ * Information and shall use it only in accordance with the terms of the
+ * license agreement you entered into with Fundacion Jala
+ */
+
+/**
+ * @author Joel Rodrigo Rojas Roman
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/compiler/CloseButton.java:9]: The author should be in the same copyright comment paragraph

File: src/main/java/org/fundacion/jala/converter/view/metadata/SelectFile.java [hunk 30]
```diff
@@ -9,13 +9,11 @@
...
-import javax.swing.JPanel;
-import javax.swing.JButton;
-import javax.swing.JTextField;
-import javax.swing.JFileChooser;
-import java.awt.Font;
-import java.awt.Dimension;
-import java.awt.FlowLayout;
+import org.fundacion.jala.converter.view.utilities.JLabelStyle;
+
+import javax.swing.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/metadata/SelectFile.java]: You should only import what is necessary, not the entire library "*"

File: src/main/java/org/fundacion/jala/converter/view/metadata/SelectFile.java [hunk 31]
```diff
@@ -9,13 +9,11 @@
...
+import javax.swing.border.EmptyBorder;
+import java.awt.*;
```
Comment (danielaSantaCruz) [inline src/main/java/org/fundacion/jala/converter/view/metadata/SelectFile.java]: You should only import what is necessary, not the entire library "*"

**Review by danielaSantaCruz — APPROVED** [review]

**Review by Raymundogs — APPROVED** [review]

**Pull request merged**